### PR TITLE
feat(packaging): add support for ipk package format

### PR DESCRIPTION
**Pull request by schmidtw**
Adds code to expose the ipk configuration values and registers the ipk package format with nfpm.

Updates the documentation with how to use the new ipk specific configuration parameters.

**This isn't ready to merge, but I have some questions**

1. I copied the `TestIPKSpecificConfig()` ([code](https://github.com/goreleaser/goreleaser/compare/main...schmidtw:goreleaser:main#diff-912a4af69daf4d89537b6bea43a1b7fe65683128ea1be66d6ec77046c76d064dR1358)) from the `TestAPKSpecificConfig()` and found it only is really testing if specific scripts are there or not.  Since IPK doesn't have any additional scripts, just fields in one file do I need this test function?  Is there a better way to validate the output?
2. I have run into issues where the tests expect `goreleaser` to be the org & repo - is there a way to override this in my fork without changing the code?
3. Is the `ToNFPAlts()` and `ToNFP()` living in the config.go file ok?  There wasn't much code in that file, so I figured I'd ask if you'd rather this code be elsewhere.

### CODE REVIEW

**Comment by schmidtw** [comment]
@caarlos0 I'm not sure if GH will ping you on a draft PR so I'm mentioning you here to get some feedback when you get some time.  Thank you!

**Review by caarlos0 — APPROVED** [review]
this looks good, thank you!

for full transparency: I plan to release v2 as 1.26 minus the deprecations, so this will probably land in v2.1 only.

I'm aiming to release v2 either this weekend or the next, depending if I get any bug reports during this time

**Comment by schmidtw** [comment]
Do you want me to update the doc to call out `v2.1` in the PR & make this a non-draft PR?

**Comment by caarlos0** [comment]
if you can yeah, I appreciate it! otherwise I can make the changes when the time comes :)

File: internal/pipe/nfpm/nfpm_test.go [hunk 1]
```diff
@@ -1340,6 +1355,76 @@ func TestAPKSpecificScriptsConfig(t *testing.T) {
 	})
 }
 
+// TODO: I need some help on advice to test the IPK specific code properly.  It
```
Comment (caarlos0) [inline internal/pipe/nfpm/nfpm_test.go:1358]: I'll take a look when I have some time

**Comment by caarlos0** [comment]
looking good!

thanks again for the pr, and for the patience 🙏

File: internal/pipe/nfpm/nfpm_test.go (hunk 1 above)
Comment (schmidtw) [reply internal/pipe/nfpm/nfpm_test.go:1358]: @caarlos0 are you waiting for a change to merge this PR in, or just getting time?

File: internal/pipe/nfpm/nfpm_test.go [hunk 2]
```diff
@@ -389,7 +389,8 @@ func TestRunPipe(t *testing.T) {
 	}
 	require.NoError(t, Pipe{}.Run(ctx))
 	packages := ctx.Artifacts.Filter(artifact.ByType(artifact.LinuxPackage)).List()
-	require.Len(t, packages, 46)
+	require.Len(t, packages, 54)
```
Comment (caarlos0) [inline internal/pipe/nfpm/nfpm_test.go:392]:
```suggestion
	require.Len(t, packages, 56)
```

File: internal/pipe/nfpm/nfpm_test.go (hunk 1 above)
Comment (caarlos0) [reply internal/pipe/nfpm/nfpm_test.go:1359]: sorry, it got lost in my notification swamp

merging it now :)

**Pull request merged**